astropy/wcs/utils.py

```diff
@@ -632,3 +583,194 @@ def pixel_to_skycoord(xp, yp, wcs, origin=0, mode='all', cls=None):
+    -----------
+    params : array
+        6 element array. First 4 elements are PC matrix, last 2 are CRPIX.
+    lon, lat: array
+        Sky coordinates.
+    u, v: array
+        Pixel coordinates
+    w_obj: `~astropy.wcs.WCS`
+        WCS object
+    """
+
+    from ..modeling.models import SIP, InverseSIP   # here to avoid circular import
+
+    # unpack params
+    crpix = params[0:2]
+    cdx = params[2:6].reshape((2, 2))
+    a_params = params[6:6+len(coeff_names)]
+    b_params = params[6+len(coeff_names):]
+
+    # assign to wcs, used for transfomations in this function
+    w_obj.wcs.pc = cdx
+    w_obj.wcs.crpix = crpix
+
+    a_coeff, b_coeff = {}, {}
+    for i in range(len(coeff_names)):
+        a_coeff['A_' + coeff_names[i]] = a_params[i]
+        b_coeff['B_' + coeff_names[i]] = b_params[i]
+
+    sip = SIP(crpix=crpix, a_order=order, b_order=order,
+              a_coeff=a_coeff, b_coeff=b_coeff)
+    fuv, guv = sip(u, v)
+
+    xo, yo = np.dot(cdx, np.array([u+fuv-crpix[0], v+guv-crpix[1]]))
+
+    x, y = w_obj.wcs_world2pix(lon, lat, 1)
+    x, y = np.dot(w_obj.wcs.pc, (x-w_obj.wcs.crpix[0], y-w_obj.wcs.crpix[1]))
+
+    resids = np.concatenate((x-xo, y-yo))
+
+    return resids
+
+
+def fit_wcs_from_points(xy, world_coords, proj_point='center', 
+                        projection='TAN', sip_distortion=True, degree=4):
+    """ Given two matching sets of coordinates on detector and sky,
+        compute the WCS. Optionally, a SIP can be fit to account for geometric
+        distortion. Returns an `~astropy.wcs.WCS` object with the best fit
+        parameters for mapping between input pixel and sky coordinates.
+    Notes
+    ------
+    - Detector (pixel) coordinates are passed in as a tuple of arrays (`xy`),
+      and sky coordinates as a `~astropy.coordinates.SkyCoord` object.
+    - The fiducial point for the spherical projection can be set to 'center'
+      to use the mean position of input sky coordinates, or as an
+      `~astropy.coordinates.SkyCoord` object.
+    - All output will be in degrees.
+    - If the coordinate frame differs between `~astropy.coordinates.SkyCoord`
+      objects passed in for `world_coords` and `proj_point`, the frame for
+      `world_coords`  will override.
+    - If `sip_distortion` is False, `degree` will be ignored. 
+    Parameters
+    ----------
+    yp : tuple of two `numpy.ndarray`
+        x & y pixel coordinates.
+    world_coords : `~astropy.coordinates.SkyCoord`
+        Skycoord object with world coordinates.
+    proj_point : 'center' or ~astropy.coordinates.SkyCoord`
+        Defaults to 'center', in which the geometric center of input world
+        coordinates will be used as the projection point. To specify an exact
+        point for the projection, a Skycoord object with a coordinate pair can
+        be passed in. For consistency, the units and frame of these coordinates
+        will be transformed to match 'world_coords' if they don't.
+    """
```

Comment:
`projection`, `sip_distortion`, and `degree` are missing from this list of parameters